### ISSUE TITLE
API spec/documentation suggestions

### ISSUE THREAD

**Issue by afeedhshaji** [onvo-ai/sdks#114]
I was just resuming the work on #8 and noticed the following:

![image](https://github.com/user-attachments/assets/a7f83c31-3403-4be2-979f-ac6b5aab62f1)

Generally, per REST standards, creating a resource should be a `POST` and updating should be `PUT` - but in this case, it's the reverse. Is that expected?

**Comment by afeedhshaji** [comment]
I also noticed the following discrepancies when testing the create automation end-point:
    - `recipient_type` is actually optional but is required per the API specs? There’s also `email_to` which seems to be required?
    - `method` field is required by the API but not mentioned in the API sec.
    - `email_body` seems to be required but is not mentioned in the API sec.

**Comment by afeedhshaji** [comment]
Also, the following for `teams` resource: the `phone_number` field is optional but mentioned as mandatory in the API spec.

**Comment by afeedhshaji** [comment]
The `sample_data` in the API spec for `datasources` is not a string and the server is returning a list object (of strings potentially).

**Comment by ronneldavis** [comment]
> Also, the following for `teams` resource: the `phone_number` field is optional but mentioned as mandatory in the API spec.

Yeah this was updated recently and the API documentation has not been updated to reflect that

**Comment by ronneldavis** [comment]
> The `sample_data` in the API spec for `datasources` is not a string and the server is returning a list object (of strings potentially).

The sample data would always return a list of objects, never strings

**Comment by ronneldavis** [comment]
> I also noticed the following discrepancies when testing the create automation end-point: - `recipient_type` is actually optional but is required per the API specs? - There’s also a `email_to` field which is optional but not documented in the API spec. - `method` field is required by the API but not mentioned in the API sec. - `email_body` seems to be required but is not mentioned in the API sec.

Yeah the automations system was updated recently as well and the API documentation is not up to date, will have to do that

**Comment by ronneldavis** [comment]
> I was just resuming the work on #8 and noticed the following:
> 
> ![image](https://private-user-images.githubusercontent.com/39899544/382596153-a7f83c31-3403-4be2-979f-ac6b5aab62f1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3ODUwMTQsIm5iZiI6MTczMDc4NDcxNCwicGF0aCI6Ii8zOTg5OTU0NC8zODI1OTYxNTMtYTdmODNjMzEtMzQwMy00YmUyLTk3OWYtYWM2YjVhYWI2MmYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA1VDA1MzE1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjOTg0OWFmOGYzOGM4ZGQ2NGM5NjQ1MDdiNjIyZDIxNGYzYjUzNTg2ODFhM2JmM2RjYjdhZTZlM2E0YjE1OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dXDIToGL5-qKp-NUHJSRXINCq-EuTRNwlrJzZaHYScs)
> 
> Generally, per REST standards, creating a resource should be a `POST` and updating should be `PUT` - but in this case, it's the reverse. Is that expected?

Yeah this was just muscle memory from the previous place I worked at that used this system. Will see if I can update it sometime soon

**Comment by afeedhshaji** [comment]
https://github.com/onvo-ai/sdks/issues/114#issuecomment-2454138514

Sorry, I meant the API spec tells it's a string but it returns a list.

**Comment by ronneldavis** [comment]
@afeedhshaji I have updated the backend to fix the PUT vs POST mismatch. Thanks for letting me know the right standard. I have updated the JS SDK to reflect the updated HTTP verbs as well.

Also I'll update the API specs with the notes above

**Comment by afeedhshaji** [comment]
Thanks. I can verify that my tests started failing now and swapping to posts/puts fixed it.
